File: Bowling/src/test/java/ch/scs/clean_coders/bowling_kata/BowlingTest.java
package ch.scs.clean_coders.bowling_kata;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Disabled;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class BowlingTest {

    private Game game;

    @BeforeEach
    void setUp() {
        game = new Game();
    }

    private void rollMany(int n, int pins) {
        for (int i = 0; i < n; ++i) {
            game.roll(pins);
        }
    }

    private void rollSpare() {
        game.roll(5);
        game.roll(5);   // spare
    }

    private void rollStrike() {
        game.roll(10);
    }

    @Test
    void gutterGame() throws Exception {
        rollMany(20, 0);
        assertEquals(0, game.score());
    }

    @Test
    void allOnes() throws Exception {
        rollMany(20, 1);
        assertEquals(20, game.score());
    }

    @Test
    void oneSpare() throws Exception {
        rollSpare();
        game.roll(3);
        rollMany(17, 0);
        assertEquals(16, game.score());
    }

    @Test
    void oneStrike() throws Exception {
        rollStrike();
        game.roll(3);
        game.roll(4);
        rollMany(16, 0);
        assertEquals(24, game.score());
    }
}

File: csharp-dotnet-bowling/src/BowlingTest/BowlingTest.cs
namespace BowlingTest;

using NUnit.Framework;
using Bowling;

public class Tests
{
    private BowlingGame g;


    [SetUp]
    public void SetUp()
    {
        g = new BowlingGame();
    }

    private void rollMany(int n, int pins)
    {
        for (int i = 0; i < n; ++i)
        {
            g.Roll(pins);
        }
    }

    private void rollSpare()
    {
        g.Roll(5);
        g.Roll(5);
    }

    private void rollStrike()
    {
        g.Roll(10);
    }

    [Test]
    public void gutterGame()
    {
        rollMany(20, 0);
        Assert.AreEqual(0, g.Score());
    }

    [Test]
    public void allOnes()
    {
        rollMany(20, 1);
        Assert.AreEqual(20, g.Score());
    }

    [Test]
    public void oneSpare()
    {
        rollSpare();
        g.Roll(3);
        rollMany(17, 0);
        Assert.AreEqual(16, g.Score());
    }

    [Test]
    public void oneStrike()
    {
        rollStrike();
        g.Roll(3);
        g.Roll(4);
        rollMany(16, 0);
        Assert.AreEqual(24, g.Score());
    }

    [Test]
    public void perfectGame()
    {
        rollMany(12, 10);
        Assert.AreEqual(300, g.Score());
    }
}